### PR TITLE
Unit and uncertainty handling when reading `pars`

### PR DESCRIPTION
**Pull request by theHenks**
This PR enables features to read and write pars with units and uncertainties from `json` files. In particular, the following functions are provided:
- `uparse(pd::PropDict)` and `ustrip(pd::PropDict)`: Parse units and strip units from a PropdDict. The units have to be stored in the `:unit` field while the value and error have to be in the `:val` and `:err` field. The parsing remove the `:val` key completely if the there is only the `:val` and `:unit` field present. The `:unit` field has to be able to be parsed by Unitful. `Ustrip` performs the opposite operation
- `measurement(pd::PropDict)` and `mstrip(pd::PropDict)`: Parse uncerainties from the `:err` field into the variable. The parsing remove the `:val` key completely if there is only the `:val` and `:err` key present.
- `readlprops` and `writelprops`: New read and write function for LEGEND `pars` which automatically reads and writes units and uncertainties.
Example input:
```
PropDict with 2 entries:
  :P00664A => PropDict(:n_tau=>5269, :sigma=>PropDict(:unit=>"μs", :val=>5.68755, :err=>0.0770621), :tau=>PropDict(:unit=>"μs", :val=>522.829, :err=>0.0855794))
  :V02166B => PropDict(:n_tau=>22106, :sigma=>PropDict(:unit=>"μs", :val=>1.88776, :err=>0.0121664), :tau=>PropDict(:unit=>"μs", :val=>431.417, :err=>0.0137728))
```
  Example output:
```
PropDict with 2 entries:
  :P00664A => PropDict(:n_tau=>5269, :sigma=>5.688±0.077 μs, :tau=>522.829±0.086 μs)
  :V02166B => PropDict(:n_tau=>22106, :sigma=>1.888±0.012 μs, :tau=>431.417±0.014 μs)
```

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -33,6 +34,7 @@ Glob = "1"
 IntervalSets = "0.6, 0.7"
 JSON = "0.21, 1"
 LRUCache = "1.5"
+Measurements = "2.11"
```
Comment (oschulz) [inline Project.toml:37]:
```suggestion
Measurements = "2"
```

Comment (theHenks) [reply Project.toml:37]: Done

File: src/filekey.jl [hunk 2]
```diff
@@ -10,6 +10,10 @@ Abstract type for data selectors like
 """
 abstract type DataSelector end
 
+# make DataSelector compatible with PropDicts
+Base.getindex(p::PropDicts.PropDict, datasel::DataSelector) = p[Symbol(datasel)]
```
Comment (oschulz) [inline src/filekey.jl:14]:
```suggestion
Base.getindex(p::PropDict, datasel::DataSelector) = p[Symbol(datasel)]
```

File: src/filekey.jl [hunk 3]
```diff
@@ -10,6 +10,10 @@ Abstract type for data selectors like
...
+Base.setindex!(p::PropDict, value, datasel::DataSelector) = setindex!(p, value, Symbol(datasel))
+Base.haskey(p::PropDicts.PropDict, datasel::DataSelector) = haskey(p, Symbol(datasel))
```
Comment (oschulz) [inline src/filekey.jl:16]:
```suggestion
Base.haskey(p::PropDict, datasel::DataSelector) = haskey(p, Symbol(datasel))
```

File: src/filekey.jl [hunk 4]
```diff
@@ -166,7 +227,7 @@ Base.convert(::Type{DataPeriod}, s::Symbol) = DataPeriod(string(s))
 
 
 """
-    DataPeriodLike = Union{DataPeriod, Integer, AbstractString}
+    DataPeriodLike = Union{DataPeriod, Symbol, AbstractString}
```
Comment (oschulz) [inline src/filekey.jl:230]:
```suggestion
    DataPeriodLike = Union{DataPeriod, Integer, Symbol, AbstractString}
```

File: src/filekey.jl [hunk 5]
```diff
@@ -223,7 +284,7 @@ Base.convert(::Type{DataRun}, s::AbstractString) = DataRun(s)
 Base.convert(::Type{DataRun}, s::Symbol) = DataRun(string(s))
 
 """
-    DataRunLike = Union{DataRun, Integer, AbstractString}
+    DataRunLike = Union{DataRun, Symbol, AbstractString}
```
Comment (oschulz) [inline src/filekey.jl:287]:
```suggestion
    DataRunLike = Union{DataRun, Integer, Symbol, AbstractString}
```

File: src/filekey.jl [hunk 6]
```diff
@@ -166,7 +227,7 @@ Base.convert(::Type{DataPeriod}, s::Symbol) = DataPeriod(string(s))
 
 
 """
-    DataPeriodLike = Union{DataPeriod, Integer, AbstractString}
+    DataPeriodLike = Union{DataPeriod, Integer, Symbol, AbstractString}
```
Comment (oschulz) [inline src/filekey.jl:230]:
```suggestion
    DataPeriodLike = Union{DataPeriod, Symbol, AbstractString}
```

File: src/filekey.jl [hunk 7]
```diff
@@ -10,6 +10,10 @@ Abstract type for data selectors like
 """
 abstract type DataSelector end
 
+# make DataSelector compatible with PropDicts
+Base.getindex(p::PropDict, datasel::DataSelector) = p[Symbol(datasel)]
```
Comment (oschulz) [inline src/filekey.jl:14]:
```suggestion
Base.getindex(p::PropDicts.PropDict, datasel::DataSelector) = p[Symbol(datasel)]
```

File: src/filekey.jl [hunk 8]
```diff
@@ -10,6 +10,10 @@ Abstract type for data selectors like
 """
 abstract type DataSelector end
 
+# make DataSelector compatible with PropDicts
+Base.getindex(p::PropDicts.PropDict, datasel::DataSelector) = p[Symbol(datasel)]
+Base.setindex!(p::PropDict, value, datasel::DataSelector) = setindex!(p, value, Symbol(datasel))
+Base.haskey(p::PropDict, datasel::DataSelector) = haskey(p, Symbol(datasel))
```
Comment (oschulz) [inline src/filekey.jl:16]:
```suggestion
Base.haskey(p::PropDicts.PropDict, datasel::DataSelector) = haskey(p, Symbol(datasel))
```

**Review by oschulz — APPROVED** [review]

**Pull request merged**